dataplane/src/nat/stateful/mod.rs

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+#![allow(unused_variables)]
+
+mod allocator;
+mod sessions;
+
+use super::Nat;
+use net::headers::Net;
+use net::vxlan::Vni;
+use std::net::IpAddr;
+
+#[derive(thiserror::Error, Debug)]
+pub enum StatefulNatError {
+    #[error("other error")]
+    Other,
+}
+
+#[derive(Debug, Clone)]
+struct NatState {}
+
+#[derive(Debug, Clone)]
+struct NatTuple {
+    src_ip: IpAddr,
+    dst_ip: IpAddr,
+    protocol: u8,
+    vni: Vni,
+}
+
+impl NatState {
+    fn new(net: &Net, pool: &allocator::NatPool) -> Self {
```

Comment:
Are we expecting to clone `net` and `pool`?  I would expect these to be transferred in as owned

Reply:
Wait I replied too fast, why should they be passed as owned? They won't be part of the generated `NatState`, they'll just be used read-only to create the new state. So by reference should be the way to go? Otherwise, I've got to clone them before passing them to the `NatState::new()` method.